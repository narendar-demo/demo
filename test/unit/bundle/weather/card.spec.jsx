import React from 'react';
import { create } from 'react-test-renderer';
import WeatherCard from '../../../../src/bundle/weather/card';

describe('WeatherCard Snapshot Test', () => {
    const weather =
    {
        dt: 1585396800,
        sunrise: 1585374769,
        sunset: 1585420943,
        temp: {
            day: 6.65,
            min: 1.27,
            max: 6.65,
            night: 1.27,
            eve: 4.46,
            morn: 6.65
        },
        feels_like: {
            day: 0.34,
            night: -4.97,
            eve: -2.43,
            morn: 0.34
        },
        pressure: 1038,
        humidity: 67,
        weather: [
            {
                id: 500,
                main: "Rain",
                description: "light rain",
                icon: "10d"
            }
        ],
        speed: 6.38,
        deg: 30,
        clouds: 52,
        rain: 0.29

    };
    it('renders correctly with all props given', () => {
        const wrapper = create(<WeatherCard weatherInfo={weather} />)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});