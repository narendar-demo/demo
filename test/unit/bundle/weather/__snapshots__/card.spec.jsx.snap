// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard Snapshot Test renders correctly with all props given 1`] = `
<div
  className="styles__Row-ohc1nz-2 styles__Card-ohc1nz-6 jqRJUU"
>
  <div
    className="styles__Row-ohc1nz-2 dagXLg"
  >
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputLabel-ohc1nz-4 iVgnfw"
      >
        28/2/2020
      </div>
    </div>
  </div>
  <div
    className="styles__Row-ohc1nz-2 dagXLg"
  >
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__Weather-ohc1nz-7 egkfMz"
        id="undefined_max"
        max="true"
      >
        7
        <div
          className="styles__TempType-ohc1nz-8 isQOer"
        >
          °C
        </div>
      </div>
      <div
        className="styles__Weather-ohc1nz-7 eKCvfN"
        id="undefined_min"
        min="true"
      >
        1
        <div
          className="styles__TempType-ohc1nz-8 isQOer"
        >
          °C
        </div>
      </div>
    </div>
  </div>
  <div
    className="styles__Row-ohc1nz-2 dagXLg"
  >
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputLabel-ohc1nz-4 iVgnfw"
      >
        Pressure
      </div>
    </div>
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputText-ohc1nz-5 hCchiC"
      >
        1038
      </div>
    </div>
  </div>
  <div
    className="styles__Row-ohc1nz-2 dagXLg"
  >
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputLabel-ohc1nz-4 iVgnfw"
      >
        Humidity
      </div>
    </div>
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputText-ohc1nz-5 hCchiC"
      >
        67
      </div>
    </div>
  </div>
  <div
    className="styles__Row-ohc1nz-2 dagXLg"
  >
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputLabel-ohc1nz-4 iVgnfw"
      >
        Desciption
      </div>
    </div>
    <div
      className="styles__Col-ohc1nz-1 fBjrqY"
    >
      <div
        className="styles__InputText-ohc1nz-5 hCchiC"
      >
        Rain
      </div>
    </div>
  </div>
</div>
`;
